### PR TITLE
[feat](flink-connector) add BATCH DELETE support

### PR DESCRIPTION
**Pull request by aihai**
## Proposed changes

add BATCH DELETE support

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [x] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: extension/flink-doris-connector/src/main/java/org/apache/doris/flink/cfg/DorisExecutionOptions.java [hunk 1]
```diff
@@ -105,6 +111,25 @@ public Builder setStreamLoadProp(Properties streamLoadProp) {
...
+        public Builder setMergeType(MergeType mergeType) {
+            this.streamLoadProp.setProperty("merge_type", mergeType.toString());
+            return this;
+        }
+
+        public Builder setDeleteLabel(String deleteLabel, String value) {
+            if (!MergeType.MERGE.toString().equals(this.streamLoadProp.getProperty("merge_type"))) {
+                return this;
+            }
+            this.streamLoadProp.setProperty("delete", deleteLabel + "=" + value);
```
Comment (morningman) [inline extension/flink-doris-connector/src/main/java/org/apache/doris/flink/cfg/DorisExecutionOptions.java:129]: the delete flag can be any expressions. For example:

`delete: ifnull(column1, 2) > 3`

So could you find a way to make it more flexible?

Comment (aihai) [reply extension/flink-doris-connector/src/main/java/org/apache/doris/flink/cfg/DorisExecutionOptions.java:129]: ok, I will check it

Comment (aihai) [reply extension/flink-doris-connector/src/main/java/org/apache/doris/flink/cfg/DorisExecutionOptions.java:129]: this.streamLoadProp.setProperty("delete", deleteflag);
How about it? If the delete flag can be any expressions, the user set the value by "deletaflag".

Comment (morningman) [reply extension/flink-doris-connector/src/main/java/org/apache/doris/flink/cfg/DorisExecutionOptions.java:129]: looks good to me~

**Pull request closed without merging**